### PR TITLE
feat(logr): add bidirectional adapter for go-logr/logr interface

### PR DESCRIPTION
**Pull request by amery**
## Summary

Add a new logr handler that provides full bidirectional adapters between slog.Logger and go-logr/logr interface:
- Use any logr implementation as a slog.Logger backend
- Use any slog.Logger as a logr.LogSink implementation

### Bidirectional Conversion
- `logr.Logger` → `slog.Logger`: Wrap logr implementations to use with slog
- `slog.Logger` → `logr.Logger`: Use slog as backend for logr-based libraries

### Implementation Details
- **Logger adapter**: Uses `internal.Loglet` for field chain management
- **Sink adapter**: Implements `logr.LogSink` with all optional interfaces
- Intelligent level mapping between slog severity and logr verbosity
- Preserves structured fields, stack traces, and logger names

### Level Mapping
- slog Debug ↔ logr V(1)
- slog Info/Warn ↔ logr V(0) (logr has no native Warn level)
- slog Error/Fatal/Panic ↔ logr Error()

### Test Coverage
- **97.3% test coverage** using `internal/testing` framework
- Comprehensive compliance test suite with bidirectional support
- Stress testing for concurrency and performance validation
- Type assertion helpers for interface validation
- Level mapping consistency tests with exception handling

## Test plan

- [x] Compliance test suite passes with bidirectional support
- [x] Level mapping tested with exceptions for Warn→Info transformation
- [x] Field preservation verified in round-trip conversions
- [x] Stress tests pass for concurrency and performance scenarios
- [x] All shared test utilities from `internal/testing` framework utilized
- [x] Edge cases tested (nil loggers, disabled loggers, panic handling)
- [x] Interface validation tests with type assertion helpers
- [x] British English spelling and CSpell dictionary compliance
- [x] Cognitive complexity optimizations applied to test functions

## Summary by CodeRabbit

* **New Features**
  * Introduced a bidirectional adapter enabling seamless interoperability between slog.Logger and go-logr/logr, allowing use of either logging system as a backend for the other.
  * Added support for structured fields, log level mapping, stack trace preservation, and named loggers in the adapter.
* **Documentation**
  * Added comprehensive documentation and usage examples for the new logging adapter.
  * Updated main README to include the new handler and clarify bidirectional adapter availability.
  * Introduced an AI Agent Development Guide for the new handler.
  * Added a LICENSE file for the handler.
* **Tests**
  * Added extensive unit tests to validate logging behavior, field handling, logger immutability, level mapping, and adapter interoperability.
  * Ensured that input maps passed to WithFields are not modified.

### CODE REVIEW

File: handlers/logr/go.mod [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+go 1.23.0
+
+require (
+	darvaza.org/core v0.17.1
+	darvaza.org/slog v0.7.6
+)
+
+require github.com/go-logr/logr v1.4.2
+
```
[the remark this answers is not in the document]
Comment (amery) [reply handlers/logr/go.mod:11]: we separate our dependencies from third party dependencies

File: handlers/logr/logr_loglet_test.go [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+				recorder.Clear()
+				logger.WithLevel(tc.level).Print("test message")
+
+				messages := recorder.GetMessages()
+				slogtest.AssertMessageCount(t, messages, 1)
+				slogtest.AssertMessage(t, messages[0], tc.expected, "test message")
+			})
+		}
+	})
+}
```
[the remark this answers is not in the document]
Comment (amery) [reply handlers/logr/logr_loglet_test.go]: check now

**Review by karasz — APPROVED** [review]
LGTM
